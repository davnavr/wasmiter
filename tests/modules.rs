#[test]
fn basic_module() {
    let wat = r#"(module
    (func (export "add_five") (param i32) (result i32)
        local.get 0
        i32.const 5
        i32.add))
"#;
    let wasm = wat::parse_str(wat).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}

#[test]
fn exception_handling() {
    let wasm = wat::parse_str(include_str!("modules/exception_handling.wat")).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}

#[test]
fn all_the_things() {
    let wasm = wat::parse_str(include_str!("modules/all_the_things.wat")).unwrap();
    insta::assert_display_snapshot!(&wasmiter::parse_module_sections(wasm.as_slice()).unwrap());
}
