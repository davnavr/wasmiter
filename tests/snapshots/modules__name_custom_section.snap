---
source: tests/modules.rs
expression: "wasmiter::parse_module_sections(wasm.as_slice()).unwrap().debug_module()"
---
[
    Ok(
        Ok(
            Type(
                [
                    FuncType {
                        parameters: [
                            Ok(
                                I32,
                            ),
                        ],
                        results: [
                            Ok(
                                I32,
                            ),
                        ],
                    },
                ],
            ),
        ),
    ),
    Ok(
        Ok(
            Function(
                [
                    Ok(
                        TypeIdx(
                            0,
                        ),
                    ),
                ],
            ),
        ),
    ),
    Ok(
        Ok(
            Export(
                [
                    Ok(
                        Export {
                            name: "mul_six",
                            kind: Function(
                                FuncIdx(
                                    0,
                                ),
                            ),
                        },
                    ),
                ],
            ),
        ),
    ),
    Ok(
        Ok(
            Code(
                [
                    Ok(
                        Func {
                            locals: [],
                            body: [
                                Ok(
                                    LocalGet(
                                        LocalIdx(
                                            0,
                                        ),
                                    ),
                                ),
                                Ok(
                                    I32Const(
                                        6,
                                    ),
                                ),
                                Ok(
                                    I32Mul,
                                ),
                                Ok(
                                    End,
                                ),
                            ],
                        },
                    ),
                ],
            ),
        ),
    ),
    Ok(
        [
            Ok(
                ModuleName(
                    "my_module_name",
                ),
            ),
            Ok(
                FunctionName(
                    [
                        Ok(
                            NameAssoc {
                                index: FuncIdx(
                                    0,
                                ),
                                name: "mul_six",
                            },
                        ),
                    ],
                ),
            ),
        ],
    ),
]
